server/config/persist_options.go

```diff
@@ -598,31 +617,54 @@ func (o *PersistOptions) CheckLabelProperty(typ string, labels []*metapb.StoreLa
 	return false
 }
 
+const ttlConfigPrefix = "/config/ttl/"
+
 // SetTTLData set temporary configuration
-func (o *PersistOptions) SetTTLData(parCtx context.Context, key string, value interface{}, ttl time.Duration) {
-	if data, ok := o.ttl[key]; ok {
-		data.Clear()
-		o.ttlCancel[key]()
+func (o *PersistOptions) SetTTLData(parCtx context.Context, client *clientv3.Client, key string, value string, ttl time.Duration) {
+	if o.ttl == nil {
+		o.ttl = cache.NewStringTTL(parCtx, time.Second*5, time.Minute*5)
 	}
-	ctx, cancel := context.WithCancel(parCtx)
-	o.ttl[key] = cache.NewStringTTL(ctx, 5*time.Second, ttl)
-	o.ttl[key].Put(key, value)
-	o.ttlCancel[key] = cancel
+	kv := clientv3.NewKV(client)
+	grantResp, _ := client.Grant(parCtx, int64(ttl.Seconds()))
```

Comment:
I think we shouldn't ignore the error here. If error happened, `SetTTLData` should return error.

Reply:
BTW, I prefer to move these KV operations in core package.

Reply:
> BTW, I prefer to move these KV operations in core package.

I think maybe we can add a `PutValue` into `etcdutil.go`